Remove optional chaining syntax from comments page initializer

djuber opened the pull request:
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Remove [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) from comments initializer code to placate the assets precompilation stage

The supported js features in the asset pipeline scripts is different from those in the webpack

## Related Tickets & Documents

https://github.com/forem/forem/pull/14018 added this code (merged earlier today)

https://github.com/forem/forem/pull/9188 sets up our custom uglifier (to enable source maps) which may be related to the syntax difference between the asset pipeline and webpack managed js files.


## QA Instructions, Screenshots, Recordings

- buildkite should build containers successfully (not error out)
- locally, asset precompilation should not error out (okay for postcss to fail, it's unrelated) 
```
RAILS_ENV=production NODE_ENV=production bundle exec rake assets:precompile
``` 


### UI accessibility concerns?

None

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: behavior neutral change to existing function
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)

fdoxyz reviewed (approved):
Pending Buildkite build passing this LGTM as a solution to the asset precompile problem when deploying to prod

aitchiss reviewed (approved):
testing the feature again, and it works as expected. I think the spec failure in cypress is actually flake 🙁 

When you load an article page, there is a split second(s) where the comment is visible before hiding:

https://user-images.githubusercontent.com/20773163/140114019-c848fb06-c55c-4c19-a4d2-2200779c9184.mp4

In the cypress test, we check for the comment to not be visible, and in some cases it's still visible when that function times out. If you force it to wait, it passes OK:

```js
  it('does not see hidden comments on an article not authored by them', () => {
    cy.visit('/admin_mcadmin/test-article-with-hidden-comments-slug');
    cy.findByText(/Some comments have been hidden by the post's author/).should(
      'exist',
    );
    cy.wait(500);
    cy.findByRole('button', { name: 'Toggle dropdown menu' }).should(
      'not.exist',
    );
  });
```

We don't want to add a hard-coded `wait`... I kinda think we don't want the flashing UI as comments are hidden either 😬  

It feels like this failing spec has less to do with the current fix, and more to do with fixing up a flaky spec, so I think my vote would be to `it.skip()` and raise an issue to look into the flashing UI / flake

EDIT - actually sometimes it's failing even with the wait... but not always. I've run through the feature locally and it seems to be hiding comments as expected as far as I can see 🤔

djuber commented:
> I think the spec failure in cypress is actually flake 

In this case I had introduced a logic error when I wrapped the condition in if(currentUser) which has been fixed - it was consistently failing before that change (also - I'm not 100% that editing code while the e2e cypress runner is active has the desired effect, I suspect my test failures were related to not reloading the cypress runner as much as they were to errors.)

I'll hammer it a few more times to make sure it's not flaking, it seemed to go solid green when I changed it.

aitchiss commented:
> In this case I had introduced a logic error when I wrapped the condition in if(currentUser) which has been fixed - it was consistently failing before that change (also - I'm not 100% that editing code while the e2e cypress runner is active has the desired effect, I suspect my test failures were related to not reloading the cypress runner as much as they were to errors.)

Somehow it managed to pass for me a few times before your fix, and I now have no idea how. You're absolutely right with the fix. 

>  I'm not 100% that editing code while the e2e cypress runner is active has the desired effect

Yes, sadly anything that's not webpacker JS requires you to restart cypress to see the change 🙁
